### PR TITLE
Fixes #158

### LINKED ISSUE #158: SSL parameters are undefined
The https server setup doesn't explicitly define safe ciphers or protocols. This means LCB could be unknowingly deployed on systems still using SSLv3 or lower.

### PR DESCRIPTION
**Pull request by geoffwhittington**

### CODE REVIEW

**Comment by geoffwhittington** [comment]
@hhaidar @sibartlett we should merge this ASAP

File: app.js [hunk 1]
```diff
@@ -37,7 +37,11 @@ var auth = require('./app/auth/index'),
 if (httpsEnabled) {
      app = express().https({
         key: fs.readFileSync(settings.https.key),
-        cert: fs.readFileSync(settings.https.cert)
+        cert: fs.readFileSync(settings.https.cert),
+        honorCipherOrder: settings.https.honorCipherOrder || true,
+        ciphers: settings.https.ciphers || 'AES128-GCM-SHA256',
+        handshakeTimeout: settings.https.handshakeTimeout || 5000,
+        secureOptions: settings.https.options || 'SSL_OP_NO_SSLv2:SSL_OP_NO_SSLv3:SSL_OP_NO_TLSv1'
```
Comment (funkaoshi) [inline app.js]: Seems like these should be default in express.oi?

**Comment by funkaoshi** [comment]
Seems like a good idea. Should rebase on master.

File: app.js [hunk 2]
```diff
@@ -37,7 +37,11 @@ var auth = require('./app/auth/index'),
 if (httpsEnabled) {
      app = express().https({
         key: fs.readFileSync(settings.https.key),
-        cert: fs.readFileSync(settings.https.cert)
+        cert: fs.readFileSync(settings.https.cert),
+        honorCipherOrder: settings.https.honorCipherOrder || true,
```
Comment (geoffwhittington) [inline app.js]: What will happen if someone sets 'false' for this setting? I'm considering just hard coding this as true. This sets whether the server (true) dictates the order of cipher selection or client (false) - we might as well leave it as true and remove the setting option.

Comment (geoffwhittington) [inline app.js]: What will happen if someone sets 'false' for this setting? I'm considering just hard coding this as true. This sets whether the server (true) dictates the order of cipher selection or client (false) - we might as well leave it as true and remove the setting option.

Comment (funkaoshi) [inline app.js]: Seems reasonable. Does this block old clients or some such thing? When would you ever want to allow the client to dictate what to use?

Comment (geoffwhittington) [inline app.js]: This shouldn't block old clients. The cipher list may cause problems but we have enough flexibility to deal with this with an update to the default yml settings.

I cant think when it is ever reasonable to let a client dictate the security of the connection.

**Comment by geoffwhittington** [comment]
This is being punted

**Pull request closed without merging**